### PR TITLE
#713 Support autoscaling on EMR

### PR DESCRIPTION
**Pull request by darrenhaken**
Can someone review this?

### CODE REVIEW

**Comment by darrenhaken** [comment]
This links to https://github.com/terraform-providers/terraform-provider-aws/issues/713

**Comment by darrenhaken** [comment]
I have added the docs now so it should be good to merge in

**Comment by darrenhaken** [comment]
#2877

**Comment by darrenhaken** [comment]
@jen20 any idea when this might get reviewed?

**Comment by bflad** [comment]
Hi @darrenhaken, thanks for your work here. Due to the fast moving nature of the AWS SDK versions, we tend to prefer vendor/dependency changes in a separate PR to prevent rebase issues like the conflicts now in this PR. Can you please rebase so it doesn't include any `vendor` changes? The master branch has v1.12.57 right now.

**Comment by darrenhaken** [comment]
@bflad should be fixed now, can you review the PR?

**Comment by darrenhaken** [comment]
@bflad has this been reviewed yet? I'd like to get cracking on another PR.

**Comment by darrenhaken** [comment]
@bflad any update on this? It's been around a month now and we'd really like this feature.

**Comment by darrenhaken** [comment]
Does *anyone* have an update on this? To go a month without any feedback doesn't fill me with motivation to keep contributing to TF 👎

**Comment by bflad** [comment]
Hi @darrenhaken, this is my fault for dropping the ball here as you've been explicitly asking me for an update (rightfully since I asked you to change something). I'm really sorry about that. 😢  We have [a lot going on in this repository](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md) and this got shuffled in with the various other issue/PR notifications.

I will give an actual review of this later today.

File: aws/resource_aws_emr_cluster.go [hunk 1]
```diff
@@ -173,6 +173,12 @@ func resourceAwsEMRCluster() *schema.Resource {
 							Optional: true,
 							Default:  0,
 						},
+						"autoscaling_policy": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
```
[the remark this answers is not in the document]
Comment (darrenhaken) [reply aws/resource_aws_emr_cluster.go]: I'll make the change now

**Comment by bflad** [comment]
Almost! `make test` is currently failing, https://travis-ci.org/terraform-providers/terraform-provider-aws/builds/340856573:

```
$ make test
==> Checking that code complies with gofmt requirements...
go test ./... -timeout=30s -parallel=4
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_emr_cluster.go:18:2: imported and not used: "github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/structure"
aws/resource_aws_emr_cluster.go:183:26: undefined: normalizeJsonString
# github.com/terraform-providers/terraform-provider-aws/aws
aws/resource_aws_emr_cluster.go:18:2: imported and not used: "github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/structure"
aws/resource_aws_emr_cluster.go:183:26: undefined: normalizeJsonString
make: *** [test] Error 2
```

Some of the active contributors/maintainers here like using [`goimports`](https://godoc.org/golang.org/x/tools/cmd/goimports) to automatically manage imports on file save. It has integrations with most editors. 😄

**Comment by darrenhaken** [comment]
@bflad oops! Thanks for the tip on `goimports`

**Comment by bflad** [comment]
LGTM 🚀 

```
Tests passed: 10
=== RUN   TestAccAWSEMRCluster_basic
--- PASS: TestAccAWSEMRCluster_basic (450.84s)
=== RUN   TestAccAWSEMRCluster_instance_group
--- PASS: TestAccAWSEMRCluster_instance_group (473.68s)
=== RUN   TestAccAWSEMRCluster_terminationProtected
--- PASS: TestAccAWSEMRCluster_terminationProtected (480.30s)
=== RUN   TestAccAWSEMRCluster_bootstrap_ordering
--- PASS: TestAccAWSEMRCluster_bootstrap_ordering (491.71s)
=== RUN   TestAccAWSEMRCluster_security_config
--- PASS: TestAccAWSEMRCluster_security_config (492.46s)
=== RUN   TestAccAWSEMRCluster_custom_ami_id
--- PASS: TestAccAWSEMRCluster_custom_ami_id (576.59s)
=== RUN   TestAccAWSEMRCluster_s3Logging
--- PASS: TestAccAWSEMRCluster_s3Logging (644.34s)
=== RUN   TestAccAWSEMRCluster_tags
--- PASS: TestAccAWSEMRCluster_tags (818.61s)
=== RUN   TestAccAWSEMRCluster_root_volume_size
--- PASS: TestAccAWSEMRCluster_root_volume_size (862.40s)
=== RUN   TestAccAWSEMRCluster_visibleToAllUsers
--- PASS: TestAccAWSEMRCluster_visibleToAllUsers (996.15s)
```

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**